src/CommonITILObject.php

```diff
@@ -8720,17 +8722,50 @@ public static function getKanbanColumns($ID, $column_field = null, $column_ids =
             $columns[$item[$column_field]]['items'][] = $card;
         }
 
-       // If no specific columns were asked for, drop empty columns.
-       // If specific columns were asked for, such as when loading a user's Kanban view, we must preserve them.
-       // We always preserve the 'No Status' column.
+        // If no specific columns were asked for, drop empty columns.
+        // If specific columns were asked for, such as when loading a user's Kanban view, we must preserve them.
+        // We always preserve the 'No Status' column.
+        $category_ids = [];
         foreach ($columns as $column_id => $column) {
             if (
                 $column_id !== 0 && !in_array($column_id, $column_ids) &&
                 (!isset($column['items']) || !count($column['items']))
             ) {
                 unset($columns[$column_id]);
+            } else if (isset($column['items'])) {
+                foreach ($column['items'] as $item) {
+                    if (isset($item['_metadata']['category'])) {
+                        $category_ids[] = $item['_metadata']['category'];
+                    }
+                }
+                //$category_ids = array_merge($category_ids, array_column($column['items'], 'category'));
+            }
+        }
+        $category_ids = array_filter(array_unique($category_ids), static function ($id) {
+            return $id > 0;
+        });
+
+        $categories = [];
+        if (!empty($category_ids)) {
+            global $DB;
+            $it = $DB->request([
```

Comment:
`ITILCategory` can be translated. When it is available, the translated name should probably be used instead of the original name.